app/test/privacy_policy/table_of_contents/toc_section_expansion_test.dart

```diff
@@ -0,0 +1,593 @@
+
+            // Will expand it automatically
+            tocController.markAsCurrentlyRead('Foo');
+            // We collapse it manually
+            tocController.toggleExpansionOfSection('Foo');
+
+            // We "scroll" around in the subchapters and the main chapter of Foo
+            tocController.markAsCurrentlyRead('Bar');
+            tocController.markAsCurrentlyRead('Baz');
+            tocController.markAsCurrentlyRead('Foo');
+            tocController.markAsCurrentlyRead('Bar');
+
+            expect(
+              tocController.currentState.sections,
+              [
+                _SectionResult('Foo', isExpanded: false),
+              ],
+            );
+          });
+        },
+      );
+      test(
+          'When manually collapsing a section that is currently read and scrolling in and out of then it will expand automatically again (default behavior)',
+          () {
+        forEveryExpansionBehavior(() {
+          final sections = [
+            _Section(
+              'Foo',
+              subsections: const [
+                _Section('Bar'),
+                _Section('Baz'),
+              ],
+            ),
+            _Section(
+              'Quz',
+              subsections: const [
+                _Section('Xyzzy'),
+              ],
+            )
+          ];
+          tocController.build(sections);
+
+          // Will expand it automatically
+          tocController.markAsCurrentlyRead('Foo');
+          // We collapse it manually
+          tocController.toggleExpansionOfSection('Foo');
+          // "Scroll" out of the Foo chapter to Quz
+          tocController.markAsCurrentlyRead('Quz');
+          // "Scroll" back to Foo
+          tocController.markAsCurrentlyRead('Foo');
+
+          expect(
+            tocController.currentState.sections,
+            [
+              _SectionResult('Foo', isExpanded: true),
+              _SectionResult('Quz', isExpanded: false),
+            ],
+          );
+        });
+      });
+      test(
+          'When manually collapsing a section that is not(!) currently read and scrolling in and out of then it will expand automatically again (default behavior)',
+          () {
+        forEveryExpansionBehavior(() {
+          final sections = [
+            _Section(
+              'Foo',
+              subsections: const [
+                _Section('Bar'),
+                _Section('Baz'),
+              ],
+            ),
+          ];
+          tocController.build(sections);
+
+          // We expand it manually
+          tocController.toggleExpansionOfSection('Foo');
+          // We collapse it manually
+          tocController.toggleExpansionOfSection('Foo');
+          // "Scroll" to Foo
+          tocController.markAsCurrentlyRead('Foo');
+
+          expect(
+            tocController.currentState.sections,
+            [
+              _SectionResult('Foo', isExpanded: true),
+            ],
+          );
+        });
+      });
```

Comment:
```suggestion
      });

```

Add new line after test.